### ISSUE TITLE
RFC - helpers for common consents

### ISSUE THREAD

**Issue by sndrs** [guardian/consent-management-platform#239]
After reviewing some TCFv2 PRs and seeing how the CMP is being used, would something like this be useful?

```js
import { getConsentFor, onConsentChange } from '@guardian/consent-management-platform`

onConsentChange(consent => {
    const ga = getConsentFor('google-analytics', consent); // true
    const comscore = getConsentFor('comscore', consent); // false

    // throws error
    const eowifnwoeifjoweinf = getConsentFor('eowifnwoeifjoweinf', consent); 

    // you can still use the consent state as now
    const complexConsentCondition = myComplexConsentTask(consent);
})
```

_EDIT 01.09.2020: Updated following https://github.com/guardian/consent-management-platform/issues/239#issuecomment-682471839_

Also, would this actually fit the consent models currently in operation?

e.g. https://github.com/guardian/support-frontend/pull/2650

**Comment by GHaberis** [comment]
This would save us having to hold a reference to the vendor IDs in our code which we're currently doing so we can interrogate  `state.tcfv2.vendorConsents` in our `onConsentChange` callback, I imagine others are doing this too, and probably in a variety of ways. 

I like that this solution provides an interface for interrogating the `state` data object, so we don't have to, this means users of the library wouldn't necessarily need to know what `state` looks like, and if that `state` model were to change in the future users of this library wouldn't need to restructure their code to deal with it, the only changes necessary would be in the library as the interface would remain the same.

**Comment by sndrs** [comment]
> This would save us having to hold a reference to the vendor IDs in our code

this feels brittle to me, def something we'd be better abstracting away. if you had cause to do something more complex you can always use the actual `__tcfapi` or `__uspapi`. but probably no one will need to do that.

**Comment by sndrs** [comment]
actually maybe it would look more like

```js
import { getConsentFor, onConsentChange } from '@guardian/consent-management-platform`

onConsentChange(consent => {
    getConsentFor('ga', consent).then(ga => {
        // ga = true
    })

    getConsentFor('comscore', consent).then(comscore => {
        // comscore = false
    })

    getConsentFor('eowifnwoeifjoweinf') // rejects
})
```

**Comment by ioanna0** [comment]
I like this refactor as it will simplify the consent management of the vendors on the listener side and encapsulate both their ids and the consent logic into the CMP library in one place. It will also be a lot easier to see the full list of vendors been used as a whole in the guardian without navigating all the repos.
It feels like a good timing to do that now, as we have already gathered all the existing vendor ids already and we won't have to do a new CMP release for every time we were gathering a new vendor id.

**Comment by paulbrown1982** [comment]
+1. As a last resort, the complex condition can be used by a client to wait for the library to pick up a new vendor.

Note: we should confirm whether we want to have 2 vendors for GA, one which applies our purposes for older GA SDKs (`ga.js` and `analytics.js`), and the current main one: (`gtag.js`) which defers to Google's own adaptor for TCFv2.

**Comment by mxdvl** [comment]
I would also keep the IAB's vendor ID beside Sourcepoint’s. This looks to me like a very good move generally, with the caveat that adding vendors will require a package release. What happens if a vendor is removed from the vendor list, would that be considered a breaking change?

One thing to note is that Sourcepoint IDs are specific to a vendor list, which I believe differs between web and apps?

**Comment by sndrs** [comment]
fixed by #276

**Comment by sndrs** [comment]
released in v6.1.0
